arangod/Aql/QueryRegistry.cpp

```diff
@@ -276,49 +280,114 @@ void QueryRegistry::closeEngine(EngineId engineId) {
           << "closing engine " << engineId << ", no query";
     }
   }
+  // Now explicitly destroy the QueryInfo before we resolve the promise,
+  // but no longer under the lock:
+  queryInfoLifetimeExtension.reset();
```

Comment:
Is it intentional that no logging and monitoring is added here?

Reply:
No I had to remove it from this place and add it in a different place. I will fix it